Improvement for scan_rate

PeterChristen577 opened the pull request:
Using ethernet-ip in Node-RED, I did some analysis of the realtime behaviour using Wireshark. There I have seen the real scan rate (320 ms) does not match the setup in the project (200 ms). Before the scan rate set up as the time delay between the last resonse of the PLC and the next request. In case of longer tag list (up to 30 tags), there are two tag groups exchanged.
Now there is a timer used that triggers the start of the request more precise.

<!--- Provide a general summary of your changes in the Title above -->

### Description, Motivation, and Context
The use of "delay" does no provide a scan_rate who meets the value setup, even if the tag list is longer as e.g. 30 tags.

## How Has This Been Tested?
PLC: CompactLogix L32E
Host 1: PC with Windows 10 and Node.js 12.16.2 (x64)
Host 2: Raspberry PI with Raspian and Node-RED v1.0.5
Ethernet packet captured with Wireshark.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/64326586/80289622-08972000-8740-11ea-83ba-5c1c69557349.png)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] This is a work in progress, and I want some feedback (If yes, please mark it in the title -> e.g. `[WIP] Some awesome PR title`)

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

PeterChristen577 committed:
Improvement for scan_rate

Using ethernet-ip in Node-RED, I did some analysis of the realtime behaviour using Wire Shark. There I have seen the real scan rate (320 ms) does not match the setup in the project (200 ms). Before the scan rate set up as the time delay between the last resonse of the PLC and the next request. In case of longer tag list (up to 30 tags), there are two tag groups exchanged.
Now there is a timer used that triggers the start of the request more precise.

PeterChristen577 committed:
Fixed coding error in line 408

not === used in comparison, comes from C coding ;)

alex-controlx commented:
scan_rate designed to create a safe delay between the scan requests. Also setting an interval on external communications is bad practice. On a poor network it can end up with number of sent requests without responses. Also if I have 1000 tags to scan then request/response time might take longer that `setInterval` delay which will cause sending another request of 1000 tags without receiving response for the first batch.
Also `timer` variable is created locally. What if I want to pause the scan out of the `scan` method.

If you need to get the responses faster, I would suggest just to reduce the delay: the `scan_rate`.

PeterChristen577 committed:
Interval for scan_rate removed

According Alex's comment I have removed the interval. I found a better solution to make the scan_rate more precise.
Now the execution time for the data exchange is measured and if any time is spare to wait, the delay is invoked.
